### ISSUE TITLE
 The History and Bookmarks sync switches are not updated immediately after a sync if they were changed on another device

### ISSUE THREAD

**Issue by heres-maria** [MozillaReality/FirefoxReality#2409]
**[Affected Versions]:**
- Firefox Reality version 7 RC5 (0706b656)

**[Affected Platforms]:**
- HTC Vive Focus Plus - Android 7.1.2 
- Oculus Quest - Android 7.1.1

**[Prerequisites]:**
- The headset is turned on and the browser is open.
- Make sure to be signed into the same FxA account on the VR device as well as another non-VR device and have them be synced.

**[Steps to reproduce]:**
1. Open the “Manage Account” card from Settings and observe the History and Bookmarks sync settings.
2. On the non-VR device, change the History and Bookmarks sync settings to other states.
3. Perform a Sync on the non-VR device.
4. Perform a Sync on the VR device and observe the History and Bookmarks sync switches.

**[Expected results]:**
- The sync switches are changed to the latest set states.

**[Actual results]:**
- The sync switches are not updated to the new states.

**[Notes]:**
- Closing and reopening the “Manage Account” card makes the switches display the proper states.
- Attached is a recording of the [issue](https://drive.google.com/file/d/1J9kcrPwba3Z1A0MS4PPwsAUVhhX1QjLE/view?usp=sharing).

**Comment by jvonitter** [comment]
I think this is a limitation of the android components. @keianhzo can you confirm?

**Comment by keianhzo** [comment]
@jvonitter We can do it. This is a duplicate of https://github.com/MozillaReality/FirefoxReality/issues/2411 which is already addressed in this PR: https://github.com/MozillaReality/FirefoxReality/pull/2586. 

To sum up: We were syncing only when the user explicitly does it or after exiting the accounts dialog but this has some side effects like #2586. I've moved this to the same way as Fenix does it, we might encounter past issues like the sync status being reverted if changing it and syncing right away but this is more similar to how Fenix works and users shouldn't be syncing that much so I guess it's better to have real time sync.

**Comment by heres-maria** [comment]
I have verified this issue and it is no longer reproducible with the latest Firefox Reality version 8 RC1 build (7a444d0c).
The buttons now first appear in the off states and change to the account states after the initial sync ends. If they are changed while the initial sync is in progress, they still change to the account states after the sync ends.
Tested on Oculus Go (Android 7.1.2), Oculus Quest (Android 7.1.1), HTC Vive Focus Plus (Android 7.1.2) and HTC Vive Focus (Android 7.1.2).
